File: ucare_frontend/src/service/receiptService.js
import axios from 'axios';
import { RECEIPT_API_BASE_URL } from './urlConfig';

class receiptService {
  
  create(data){
    return axios.post(RECEIPT_API_BASE_URL + '/create', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieveAll(patientNo){
    return axios.get(RECEIPT_API_BASE_URL + '/retrieveAll/' + patientNo, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  updateState(data){
    return axios.put(RECEIPT_API_BASE_URL + '/updateState', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  delete(receiptNo){
    return axios.delete(RECEIPT_API_BASE_URL + '/delete/' + receiptNo, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  createRev(data){
    return axios.post(RECEIPT_API_BASE_URL + '/createRev', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieveOverlap(receiptOverlap){
    return axios.get(RECEIPT_API_BASE_URL + '/retrieveOverlap/' + receiptOverlap, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieveByReceiptNo(receiptNo){
    return axios.get(RECEIPT_API_BASE_URL + '/retrieveByReceiptNo/' + receiptNo, { headers: { Authorization: localStorage.getItem("authorization") }});
  }
  
}

export default new receiptService();

File: ucare_backend/src/main/java/com/douzone/ucare/service/ReceiptService.java
package com.douzone.ucare.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.douzone.ucare.repository.ReceiptRepository;
import com.douzone.ucare.repository.ReservationRepository;
import com.douzone.ucare.vo.ReceiptVo;
import com.douzone.ucare.vo.ReservationVo;

@Service
public class ReceiptService {
	
	@Autowired
	private ReceiptRepository receiptRepository;
	
	@Autowired
	private ReservationRepository reservationRepository;

	public int create(ReceiptVo receipt) {
		List<ReceiptVo> receiptList = receiptRepository.retrieveByDuplication(receipt.getPatientNo());
		List<ReservationVo> reservationList = reservationRepository.retrieveByPatientNo(receipt.getPatientNo());
		
		if(receiptList.size() != 0 || reservationList.size() != 0) 
			return 0;
		else 
			return receiptRepository.create(receipt);
	}

	public List<ReceiptVo> retrieveAll(Long patientNo) {
		return receiptRepository.retrieveAll(patientNo);
	}
	
	public int updateState(ReceiptVo receipt) {
		return receiptRepository.updateState(receipt);
	}
	
	public int delete(Long receiptNo) {
		return receiptRepository.delete(receiptNo);
	}
	
	public int createRev(ReceiptVo receipt) {
		return receiptRepository.createRev(receipt);
	}
	
	public ReceiptVo retrieveByReceiptNo(Long receiptNo) {
		return receiptRepository.retrieveByReceiptNo(receiptNo);
	}
	
}